javascript-source/init.js

```diff
@@ -512,34 +512,44 @@
                 consoleDebug('Command !' + command + ' was not sent due to the user not having permission for it.');
                 return;
-            } else
+            }
 
             // Check the command cost.
             if ($.priceCom(sender, command, subCommand, isMod) === 1) {
                 $.sayWithTimeout($.whisperPrefix(sender) + $.lang.get('cmd.needpoints', $.getPointsString($.getCommandPrice(command, subCommand, ''))), $.getIniDbBoolean('settings', 'priceComMsgEnabled', false));
                 consoleDebug('Command !' + command + ' was not sent due to the user not having enough points.');
                 return;
-            } else
-                // Check the command cooldown.
-                var oncooldown = false;
-            if (args.length > 1 && $.coolDown.exists(command + ' ' + args[0] + ' ' + args[1])) {
-                oncooldown = $.coolDown.get(command + ' ' + args[0] + ' ' + args[1], sender, isMod) !== 0;
-            } else if (args.length > 0 && $.coolDown.exists(command + ' ' + args[0])) {
-                oncooldown = $.coolDown.get(command + ' ' + args[0], sender, isMod) !== 0;
-            } else {
-                oncooldown = $.coolDown.get(command, sender, isMod) !== 0;
             }
 
-            if (oncooldown) {
-                $.sayWithTimeout($.whisperPrefix(sender) + $.lang.get('init.cooldown.msg', command, $.coolDown.getSecs(sender, command, isMod)), $.getIniDbBoolean('settings', 'coolDownMsgEnabled', false));
-                consoleDebug('Command !' + command + ' was not sent due to it being on cooldown.');
+            // Check the command cooldown.
+            var cooldownDuration,
+                isGlobalCooldown,
+                cooldownCommand = command;
+
+            if (args.length === 1 && $.coolDown.exists(cooldownCommand + ' ' + args[0])) {
+                cooldownCommand += ' ' + args[0];
+            }
+            if (args.length > 1 && $.coolDown.exists(cooldownCommand + ' ' + args[1])) {
+                cooldownCommand += ' ' + args[1];
+            } 
+
+            [cooldownDuration, isGlobalCooldown] = $.coolDown.get(cooldownCommand, sender, isMod);
+
+            if (cooldownDuration > 0) {
+                if (isGlobalCooldown) {
+                    $.sayWithTimeout($.whisperPrefix(sender) + $.lang.get('init.cooldown.msg.global', command, cooldownDuration), $.getIniDbBoolean('settings', 'coolDownMsgEnabled', false));
+                    consoleDebug('Command ! ' + command + ' was not sent due to it being on a global cooldown.');
+                } else {
+                    $.say($.whisperPrefix(sender) + $.lang.get('init.cooldown.msg.user', command, cooldownDuration), $.getIniDbBoolean('settings', 'coolDownMsgEnabled', false));
```

Comment:
$.say does not have a second parameter

$.sayWithTimeout blocks the bot from sending the message if $.sayWithTimeout has sent a message within the last 10 seconds

The second parameter in $.sayWithTimeout was added by someone who was lazy to replace a normal `if (thisOutputIsEnabled)`

Either replace $.say with $.sayWithTimeout or remove the second parameter and just do a normal if block

Reply:
I left the `$.sayWithTimeout` on purpose in the case of a global cooldown due to the fact that the bot can potentially get spammy:
User 1: !age
Bot: Standard output message
User 2: !age
Bot: Global cooldown message
User 3: !age
Bot: Global cooldown message

Using `$.sayWithTimeout` elevates this potential spam problem. Though if you prefer to also just use `$.say` here I'll change it.

On the other hand I will still need to check if the "Send Cooldown Message" option is enabled, currently the per-user cooldown message is sent even without the option not being turned on.